ionic-angular-template/src/services/auth.service.ts

```diff
@@ -35,18 +35,21 @@ export class AuthService {
 
   signIn() {
     return this.api.get(`${environment.apiUrl}/users/wallet`);
-}
+  }
+
+  async createUser(wallet: string, uid: string, email: string) {
+    return this.api.post(`${environment.apiUrl}/users/createUser`, {wallet, uid, email}).toPromise();
+
+  }
 
   async signUp(email: string, password: string) {
     try {
       const currentUser = await firebase.auth().createUserWithEmailAndPassword(email, password);
       await currentUser.user.sendEmailVerification();
       const wallet = ethers.Wallet.createRandom();
       const encryptPromise = await wallet.encrypt(password);
-      return this.api.post(`${environment.apiUrl}/users/createUser`, {wallet: encryptPromise, uid: currentUser.user.uid, email}).pipe(tap((user: any) => {
-        // this.storageService.setItem('user', JSON.stringify(currentUser.user));
-        this.loggedUserDataSubject$.next(user.user);
-      }));
+      const user: any = await this.createUser(encryptPromise, currentUser.user.uid, email);
```

Comment:
remove that "any"

Reply:
I got error because that variable 'user' is type of ArrayBuffer.